src/ucs/config/parser.c

```diff
@@ -2138,41 +2153,37 @@ ucs_config_parser_print_opts_recurs(FILE *stream, const void *opts,
 
                 head = ucs_list_head(prefix_list, ucs_config_parser_prefix_t, list);
 
-                ucs_config_parser_print_field(stream,
-                                              UCS_PTR_BYTE_OFFSET(opts, alias_table_offset),
-                                              prefix, prefix_list,
-                                              field->name, aliased_field,
-                                              flags, "%-*s %s%s%s",
-                                              UCS_CONFIG_PARSER_DOCSTR_WIDTH,
-                                              "alias of:", prefix,
-                                              head->prefix,
-                                              aliased_field->name);
+                ucs_config_parser_print_field(
+                        stream, UCS_PTR_BYTE_OFFSET(opts, alias_table_offset),
+                        prefix, prefix_list, field->name, aliased_field, flags,
+                        title, printed, filter, "%-*s %s%s%s",
+                        UCS_CONFIG_PARSER_DOCSTR_WIDTH, "alias of:", prefix,
+                        head->prefix, aliased_field->name);
             }
         } else {
             if (ucs_config_is_deprecated_field(field) &&
                 !(flags & UCS_CONFIG_PRINT_HIDDEN)) {
                 continue;
             }
             ucs_config_parser_print_field(stream, opts, prefix, prefix_list,
-                                          field->name, field, flags, NULL);
+                                          field->name, field, flags, title,
+                                          printed, filter, NULL);
         }
     }
 }
 
-void ucs_config_parser_print_opts(FILE *stream, const char *title, const void *opts,
-                                  ucs_config_field_t *fields, const char *table_prefix,
-                                  const char *prefix, ucs_config_print_flags_t flags)
+void ucs_config_parser_print_opts(FILE *stream, const char *title,
+                                  const void *opts, ucs_config_field_t *fields,
+                                  const char *table_prefix, const char *prefix,
+                                  ucs_config_print_flags_t flags,
+                                  const char *filter)
 {
+    size_t printed = 0;
     ucs_config_parser_prefix_t table_prefix_elem;
     UCS_LIST_HEAD(prefix_list);
 
-    if (flags & UCS_CONFIG_PRINT_DOC) {
-        fprintf(stream, "# UCX library configuration file\n");
-        fprintf(stream, "# Uncomment to modify values\n");
-    }
-
-    if (flags & UCS_CONFIG_PRINT_HEADER) {
-        fprintf(stream, "\n");
+    if ((flags & UCS_CONFIG_PRINT_HEADER) &&
```

Comment:
1. maybe add common function to print the header? and use it also from ucs_config_parser_print_opts_recurs()
2. put this block as the "else" section of the `if (flags & UCS_CONFIG_PRINT_CONFIG) {` of line 2193